external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -1439,6 +1439,7 @@ fn parse_sequence(context: &mut Context) -> Result<Sequence, Box<Diagnostic>> {
     let mut seq: Vec<SequenceItem> = vec![];
     let mut last_semicolon_loc = None;
     let mut eopt = None;
+    let mut parsing_error = false;
```

Comment:
Understood here but it feels a bit strange. For the errors we had in question, I would have imagined `parse_sequence_item` to have returned an UnresolvedError?

Reply:
> I would have imagined parse_sequence_item to have returned an UnresolvedError?

This would indeed be preferred, but I'd rather not go into this rabbit hole just yet. The reason it's a bit complicated is that on failure in parsing `SequenceItem`, we need to fast forward the parser to the end of the item (so that parsing at `Sequence` level does not produce a bunch of weird errors trying to restart parsing). The stop set for sequence item set at the `Sequence` level would include `;` and `}` but we can't simply stop at `}` if we are parsing something like this:
```
    fun test00(): S<u64> {
        0x42<u64>::m::S { u: 0 }
    }
```

If we do, we will stop at `}` finishing pack rather than at the `}` finishing sequence. What we'd need to do is to count opening and closing braces, but at the level we need to do it, we technically don't know that it should be done (as we only have a stop set as the source of information).

Currently we handle this by having an error returned from `SequenceItem` parsing and doing the correct fast-forwarding at `SequenceLevel` where we know how `SequenceItem` can end and what tokens to look for.

Reply:
Why can't we just have `match parse_sequence_item(context) { ... | Err(diag) => {     seq.push(UnresolvedError); ...`? Is more my point

Reply:
Doh... Now that I tried it, though, I realized that it wouldn't quite work. The reason for it is that in order to suppress the typing error for non-parsable function body, there can only be one sequence element and this element must be `UnresolvedError`. Unfortunately, if we place `UnresolvedError` in `seq` here and the last no-semicolon statement is empty, expansion's translation pass will insert another element to the sequence for the non-semicolon statement.

I can still insert `UnresolvedError` into the sequence and at the end move it to no-semicolon placeholder if the placeholder is empty and there is only one list element, but this arguably wouldn't be much prettier...

Added a comment along those lines, but open to additional suggestions.

Reply:
Actually, on the second thought, perhaps translation to expansion should be changed to not produce a final unit-typed expression if the only sequence member is unresolved. I made a change along those lines, please let me know what you think.